app/controllers/application_controller/filter/expression.rb

```diff
@@ -453,7 +453,9 @@ def update_from_exp_tree(exp)
       end
       if val2 && [:datetime, :date].include?(val2[:type])
         self.exp_cvalue = Array.wrap(exp_cvalue)              # Turn date/time cvalues into an array
-        val2[:date_format] = exp_cvalue.first.include?('/') ? 's' : 'r'
+        if exp_cvalue.first
+          val2[:date_format] = exp_cvalue.first.include?('/') ? 's' : 'r'
+        end
         if ckey == EXP_FROM && val2[:date_format] == 'r'
```

Comment:
That's an excellent question, but since 456 is intended merely as a presence check, I think the question you asked is probably better suited to a different PR as it's a separate concern from fixing the bug.

Reply:
so if it is a date or time, seems strange that it is comparing the value to a nil

If we default to setting to an `r`, then that means we will assign format to `r` for `exp_cvalue.nil?` and the `:through_choices` will be passed a nil - which seems possible wrong (not sure)

Reply:
followup:

https://github.com/manageiq/manageiq-ui-classic/blob/master/app/controllers/application_controller/filter/expression.rb#L525

```ruby
    def process_datetime_expression_field(value_key, exp_key, exp_value_key)
      if [:date, :datetime].include?(self[value_key][:type]) # Seting value for date/time fields
        self[value_key][:date_format] ||= 'r'
```

looks like we do want to default to `r`, but we need to ensure that we handle nil correctly here